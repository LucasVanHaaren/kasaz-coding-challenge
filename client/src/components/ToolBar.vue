<template>
  <v-toolbar color="purple darken-2" dark>
    <v-text-field
      solo
      light
      background-color="white"
      color="green"
      flat
      hide-details
      placeholder="Search"
      dense
      clearable
      v-model="searchBarData"
    />
    <v-btn icon color="green" @click.native="emitSearchBarClick">
      <v-icon>fas fa-search</v-icon>
    </v-btn>
    <v-spacer v-if="$vuetify.breakpoint.mdAndUp"></v-spacer>
    <v-btn text @click.native="emitFiltersClick">
      Filtros
      <v-icon v-if="showFilters" color="green" right>fas fa-chevron-up</v-icon>
      <v-icon v-else color="green" right>fas fa-chevron-down</v-icon>
    </v-btn>
  </v-toolbar>
</template>

<script>

export default {
  name: 'ToolBar',
  data: () => ({
    searchBarData: ""
  }),
  methods: {
    emitFiltersClick: function() {
      this.$emit("filtersClick");
    },
    emitSearchBarClick: function() {
      if(this.searchBarData) {
        this.$emit("searchBarClick", this.searchBarData);
        this.searchBarData = "";
      }
    }
  },
  props: {
    showFilters: {
      type: Boolean,
      required: true
    }
  }
};
</script>