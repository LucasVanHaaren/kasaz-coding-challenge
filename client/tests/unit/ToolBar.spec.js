import { shallowMount, mount } from '@vue/test-utils'
import ToolBar from '@/components/ToolBar.vue'

describe('ToolBar.vue', () => {

  it('match snapshot', () => {
    const wrapper = shallowMount(ToolBar, {
      propsData: {
        showFilters: false,
        showClearFiltersBtn: false
      },
      mocks: {
        $vuetify: {
          breakpoint: {
            mdAndUp: true
          } 
        }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders icon correctly', () => {
    const wrapper = shallowMount(ToolBar, {
      propsData: {
        showFilters: false,
        showClearFiltersBtn: false
      },
      mocks: {
        $vuetify: {
          breakpoint: {
            mdAndUp: true
          } 
        }
      }
    });
    expect(wrapper.find('v-icon').isVisible()).toBe(true);
  });

})
