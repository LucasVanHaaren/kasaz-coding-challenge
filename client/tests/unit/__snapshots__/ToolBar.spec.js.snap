// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBar.vue match snapshot 1`] = `
<v-toolbar
  color="purple darken-2"
  dark=""
>
  <v-text-field
    background-color="white"
    clearable=""
    color="green"
    dense=""
    flat=""
    hide-details=""
    light=""
    placeholder="Buscar"
    solo=""
  />
   
  <v-btn
    color="green"
    icon=""
  >
    <v-icon>
      fas fa-search
    </v-icon>
  </v-btn>
   
  <v-spacer />
   
  <!---->
   
  <v-btn
    text=""
  >
    
    Filtros
    
    <v-icon
      color="green"
      right=""
    >
      fas fa-chevron-down
    </v-icon>
  </v-btn>
</v-toolbar>
`;
